### ISSUE TITLE
Autobot training problems

### ISSUE THREAD

**Issue by ChengkaiYang** [vita-epfl/UniTraj#6]
Dear Author,
       In autobot model codes provided by your team, to avoid any row of key padding mask is True, road_pts_mask is deal with following operation:
      road_pts_mask[:, 0][road_pts_mask.sum(-1) == roads.shape[2]] = False
      I wonder why not do the same operation with road segment mask? We do the multihead attention regarding latent variables as query, road features as key, road_segment_mask as key padding mask. It can't be guaranteed that it will never appear nan.
     So I suggest to fix the code as this:
![image](https://github.com/vita-epfl/UniTraj/assets/71952712/941302ce-2e29-4290-9646-ad253e9a291c)
     Thank you so much!

**Comment by Alan-LanFeng** [comment]
Hi,

Thank you for the careful review of the code! The suggestion is reasonable and effective. Can you please create a pull request so I can merge the change? Thanks.

**Comment by ChengkaiYang** [comment]
> Hi,
> 
> Thank you for the careful review of the code! The suggestion is reasonable and effective. Can you please create a pull request so I can merge the change? Thanks.

Thanks for your reply!But i feel so sorry that I'd love to help you but I'm new to github.To fix that latent bug, i only add one line to the origin code that:
road_segment_mask[torch.all(road_segment_mask, dim=-1), 0] = False
So it can be easily added to your excellent work.
Maybe, I also find some possible questions in your codes when following your excellent work:
1) In scenarionet code, i found a latent small bug when i am preprocessing Argoverse2 dataset.
After my check, the bug occurs because when dealing with position x coordinate, we use variable "step_count" as a slice to obj_state["state"]["position"].Some case it is 0.0,but slice can't be float or int.So i made following change to change the type of "step_count" to int in file"scenarionet/converter/argoverse2/utils.py" at line 70 in the function "extract_tracks":
![image](https://github.com/vita-epfl/UniTraj/assets/71952712/376c75de-eff4-4521-86d7-2609181f05f2)
2) Your code have use pytorch lightning. In autobot's code, when doing transformer decoder layer, we set batch_first=False,but pytorch lightning has raise a warning to suggest use batch_first when enable_nested_tensor is True so that it can get better performance. So maybe we can make batch first or set enable_nested_tensor=False in "social_attn_layers" and "temporal_attn_layers". It will remove the possible warning.
3) When set the strategy="auto" in pl.trainer, it can not find "auto" and return an error, so maybe i can only set "auto" to "ddp".
4) The generalization ability of training autobot at Argoverse2 and evaluating at Argoverse2. In your paper, the brier_fde6 result is 2.51.But I run the autobot at Argoverse2 and find the result can at least be about 1.85. I use StratifiedShuffleSplit method to split Argoverse2 to make the rate of 8 type trajectory(stationary, straight,left_turn...) in train set and validation set is equal and get the result.
![image](https://github.com/vita-epfl/UniTraj/assets/71952712/21bb182f-b5e0-4dff-9c0f-c89237449302)

Thank you for your patience!

**Comment by Alan-LanFeng** [comment]
Thank you very much for pointing out all these issues. We really appreciate your contribution to this framework and will fix these potential problems asap.

By the way, I'm curious about 4: your results on AV2 look really good! Did you get this result on the official validation set or the validation set split by 'StratifiedShuffleSplit'?

**Comment by ChengkaiYang** [comment]
> Thank you very much for pointing out all these issues. We really appreciate your contribution to this framework and will fix these potential problems asap.
> 
> By the way, I'm curious about 4: your results on AV2 look really good! Did you get this result on the official validation set or the validation set split by 'StratifiedShuffleSplit'?

I got the result on the validation set split by 'StratifiedShuffleSplit' using 80000 cases in AV2.I split train set and validation set by the rate 0.85:0.15.The split is conducted through using following codes:
StratifiedShuffleSplit(n_splits=1, test_size=0.15, train_size=0.85, random_state=cfg.seed)

**Comment by ChengkaiYang** [comment]
> Thank you very much for pointing out all these issues. We really appreciate your contribution to this framework and will fix these potential problems asap.
> 
> By the way, I'm curious about 4: your results on AV2 look really good! Did you get this result on the official validation set or the validation set split by 'StratifiedShuffleSplit'?

About the result on the official validation set: Although i haven't submit the result of autobot to the official benchmark of AV2.But i check the benchmark and find the sota has been 1.65. Consider that autobot is published in 2022,I thought autobot's result might not be higher than 1.65.

**Comment by ChengkaiYang** [comment]
> Thank you very much for pointing out all these issues. We really appreciate your contribution to this framework and will fix these potential problems asap.
> 
> By the way, I'm curious about 4: your results on AV2 look really good! Did you get this result on the official validation set or the validation set split by 'StratifiedShuffleSplit'?

By the way, i found scenarionet has provided us to split train set and validation set automatically.But i am not sure if scenarionet has considered the imbalanced distributions about the 8 trajectory type when spliting the whole dataset?So i didn't use the method provided by scenarionet when spliting the whole dataset into train set and validation set.
![image](https://github.com/vita-epfl/UniTraj/assets/71952712/9e52e3a1-7a6f-43d8-a587-2c056b1e8561)

**Comment by Alan-LanFeng** [comment]
> > Thank you very much for pointing out all these issues. We really appreciate your contribution to this framework and will fix these potential problems asap.
> > By the way, I'm curious about 4: your results on AV2 look really good! Did you get this result on the official validation set or the validation set split by 'StratifiedShuffleSplit'?
> 
> By the way, i found scenarionet has provided us to split train set and validation set automatically.But i am not sure if scenarionet has considered the imbalanced distributions about the 8 trajectory type when spliting the whole dataset?So i didn't use the method provided by scenarionet when spliting the whole dataset into train set and validation set. ![image](https://private-user-images.githubusercontent.com/71952712/335630081-9e52e3a1-7a6f-43d8-a587-2c056b1e8561.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcyMzMxNjQsIm5iZiI6MTcxNzIzMjg2NCwicGF0aCI6Ii83MTk1MjcxMi8zMzU2MzAwODEtOWU1MmUzYTEtN2E2Zi00M2Q4LWE1ODctMmMwNTZiMWU4NTYxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjAxVDA5MDc0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk1ZmIyNzNkNzc1NTFjN2Q1OGEwOWY5MGM3OTg5MzM0ODgyOTQyNWVlOTdjMmE0NmU4OWViZWZkZmQ5YWI5NjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8uEJ4LizOViaphRTWLr9Kv90RbPxKv50dT7nUlEMtqg)

ScenarioNet splits the dataset based on its official split (A training set and a validation set).
